item-impl/src/main/java/com/example/auction/item/impl/PItem.java

```diff
@@ -12,32 +12,35 @@
 
     private final UUID id;
     private final UUID creator;
-    private final String title;
-    private final String description;
-    private final String currencyId;
-    private final int increment;
-    private final int reservePrice;
+    private final PItemDetails itemDetails;
     private final int price;
     private final PItemStatus status;
-    private final Duration auctionDuration;
     private final Optional<Instant> auctionStart;
     private final Optional<Instant> auctionEnd;
     private final Optional<UUID> auctionWinner;
 
     @JsonCreator
+    private PItem(UUID id, UUID creator, PItemDetails pItemDetails, int price, PItemStatus status,
+                  Optional<Instant> auctionStart, Optional<Instant> auctionEnd, Optional<UUID> auctionWinner) {
+        this.id = id;
+        this.creator = creator;
+        this.itemDetails = pItemDetails;
+        this.price = price;
+        this.status = status;
+        this.auctionStart = auctionStart;
+        this.auctionEnd = auctionEnd;
+        this.auctionWinner = auctionWinner;
+    }
+
+    @Deprecated
```

Comment:
Why deprecate? It's an implementation class, not part of the API, so could just go ahead and remove it.